climada/util/tag.py

```diff
@@ -0,0 +1,121 @@
+"""
+This file is part of CLIMADA.
+
+Copyright (C) 2017 ETH Zurich, CLIMADA contributors listed in AUTHORS.
+
+CLIMADA is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free
+Software Foundation, version 3.
+
+CLIMADA is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with CLIMADA. If not, see <https://www.gnu.org/licenses/>.
+
+---
+
+Define Tag class.
+"""
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Union, List
+import h5py
+
+__all__ = ['Tag']
+
+
+STR_DT = h5py.special_dtype(vlen=str)
+
+
+def _distinct_list_of_str(list_of_str: list, arg: Union[list, str, object]):
+    if arg:
+        if isinstance(arg, list):
+            for itm in arg:
+                if str(itm) not in list_of_str:
+                    list_of_str.append(str(itm))
+        else:
+            list_of_str.append(str(arg))
+    return list_of_str
+
+
+class Tag():
```

Comment:
In another file, there is an equality comparison for tags. Should then the `__eq__` not be defined?

Reply:
Good point! And actually yes, Tag is just the kind of object/class where implementing `__eq__` would make a lot of sense, imho.
However - since Tag is to be removed altogether, I guess the effort is not justified. 
(The effort of implementing `__eq__` may be small enough but finding the places where it can or should be used seems to be serious work.)

Reply:
😲 Er - which equality comparison?